lib/Tool.php

```diff
@@ -328,6 +329,29 @@ public static function isFrontendRequestByAdmin(Request $request = null)
             ->isFrontendRequestByAdmin($request);
     }
 
+    /**
+     * Verify element request (eg. editmode, preview, version preview) called within admin, with permissions.
+     *
+     * @param Request $request
+     * @param Element\ElementInterface $element
+     *
+     * @return bool
+     */
+    public static function isElementRequestByAdmin(Request $request, Element\ElementInterface $element)
+    {
+        if (!self::isFrontendRequestByAdmin($request)) {
+            return false;
+        }
+
+        $user = Tool\Authentication::authenticateSession($request);
+
+        if (!($user && $element->isAllowed('view', $user))) {
+            return false;
+        }
+
+        return true;
```

Comment:
For me, it's more clear when it's written like this:

```suggestion
        return $user && $element->isAllowed('view', $user);
```